scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -443,12 +443,13 @@ def monitor_serial(self, ser, halt_event, harness):
             # Just because ser_fileno has data doesn't mean an entire line
             # is available yet.
             if serial_line:
-                sl = serial_line.decode('utf-8', 'ignore').lstrip()
-                logger.debug("DEVICE: {0}".format(sl.rstrip()))
-
-                log_out_fp.write(strip_ansi_sequences(sl).encode('utf-8'))
-                log_out_fp.flush()
-                harness.handle(sl.rstrip())
+                # can be more lines in serial_line so split them before handling
```

Comment:
as I understand the problem to solve here is because `serial_line()` expects `\n`, but for some reason (it might be utf-8 decoding ? and/or running tests on windows ?) `\r\n` becomes `\r\xff`. The solution proposed is to rely on `str.splitlines()` implementation which uses 'extended universal newlines' boundaries including `\r'.

Reply:
> as I understand the problem to solve here is because serial_line() expects \n

no. I modified Twister a bit to catch that log and print raw data. `\xff` is removed by `decode('utf-8', 'ignore')`.
And for `splitlines` there is no matter if there is \n, \r, \r\n ... all are treated as new lines. For unit test I wanted to keep it simple (and to not cause problems when executing it on Windows), and to just test if lines will be handled one by one.

Reply:
thanks for confirming my understanding